src/System.Memory/tests/Span/Overflow.cs

```diff
@@ -60,6 +60,219 @@ public static void IndexOverflow()
+            slice[1] = guid;
+
+            Assert.Equal(guid, arr[3]);
+        }
+
+        [Fact]
+        public static void ReadOnlySliceStartInt32Overflow()
+        {
+            IntPtr huge;
+            try
+            {
+                huge = Marshal.AllocHGlobal(new IntPtr(ThreeGiB));
+            }
+            catch (Exception)
+            {
+                return;  // It's not implausible to believe that a 3gb allocation will fail - if so, skip this test to avoid unnecessary test flakiness.
+            }
+
+            try
+            {
+                unsafe
+                {
+                    int
+                        GuidThreeGiBLimit = (int)(ThreeGiB / sizeof(Guid)),
+                        GuidTwoGiBLimit = (int)(TwoGiB / sizeof(Guid)),
+                        GuidOneGiBLimit = (int)(OneGiB / sizeof(Guid));
+
+                    var mutable = new Span<Guid>((void*)huge, GuidThreeGiBLimit);
+                    var span = new ReadOnlySpan<Guid>((void*)huge, GuidThreeGiBLimit);
+                    Guid guid = Guid.NewGuid();
+                    var slice = span.Slice(GuidTwoGiBLimit + 1);
+                    mutable[GuidTwoGiBLimit + 1] = guid;
+                    Assert.Equal(guid, slice[0]);
+
+                    slice = span.Slice(GuidOneGiBLimit).Slice(1).Slice(GuidOneGiBLimit);
+                    Assert.Equal(guid, slice[0]);
+                }
+            }
+            finally
+            {
+                Marshal.FreeHGlobal(huge);
+            }
+        }
+
+        [Fact]
+        public static void ReadOnlySliceStartUInt32Overflow()
+        {
+            IntPtr huge;
+            try
+            {
+                huge = Marshal.AllocHGlobal(new IntPtr(ThreeGiB));
+            }
+            catch (Exception)
+            {
+                return;  // It's not implausible to believe that a 3gb allocation will fail - if so, skip this test to avoid unnecessary test flakiness.
+            }
+
+            try
+            {
+                unsafe
+                {
+                    int
+                        GuidThreeGiBLimit = (int)(ThreeGiB / sizeof(Guid)),
+                        GuidTwoGiBLimit = (int)(TwoGiB / sizeof(Guid)),
+                        GuidOneGiBLimit = (int)(OneGiB / sizeof(Guid));
+
+                    var mutable = new Span<Guid>((void*)huge, GuidThreeGiBLimit);
+                    var span = new ReadOnlySpan<Guid>((void*)huge, GuidThreeGiBLimit);
+                    Guid guid = Guid.NewGuid();
+                    var slice = span.Slice((GuidTwoGiBLimit + 1));
+                    mutable[GuidTwoGiBLimit + 1] = guid;
+                    Assert.Equal(guid, slice[0]);
+
+                    slice = span.Slice(GuidOneGiBLimit).Slice(1).Slice(GuidOneGiBLimit);
+                    Assert.Equal(guid, slice[0]);
+                }
+            }
+            finally
+            {
+                Marshal.FreeHGlobal(huge);
+            }
+        }
```

Comment:
What's the difference with all of these "huge" tests? Why do we need them all?

Reply:
ReadOnlySliceStartUInt32Overflow and ReadOnlySliceStartLengthInt32Overflow are essentially identical. I don't think we need both.